apiserver/facades/agent/provisioner/service.go

```diff
@@ -63,6 +63,12 @@ type MachineService interface {
 	// exists.
 	// It returns a NotFound if the given machine doesn't exist.
 	SetKeepInstance(ctx context.Context, machineName coremachine.Name, keep bool) error
+	// SetMachineCloudInstance sets an entry in the machine cloud instance table
+	// along with the instance tags and the link to a lxd profile if any.
+	SetMachineCloudInstance(ctx context.Context, machineUUID string, instanceID instance.Id, hardwareCharacteristics *instance.HardwareCharacteristics) error
+	// GetMachineUUID returns the UUID of a machine identified by its name.
+	// It returns a MachineNotFound if the machine does not exist.
+	GetMachineUUID(ctx context.Context, name coremachine.Name) (string, error)
```

Comment:
Can we rename this to `GetMachineID` and return a new machine.ID. Similar to how charm is done.

We can do it in a subsequent PR to keep this one focused, but that's the direction we're going now.